utils/global-state-update-gen/src/validators/validators_manager.rs

```diff
@@ -83,6 +93,38 @@ impl ValidatorsUpdateManager {
         self.add_and_remove_bids(&validators_diff, &new_snapshot);
 
         self.remove_withdraws(&validators_diff);
+
+        // Create accounts for non-validators (0 stake) - validators have been taken care of when
+        // creating bids.
+        // If the account already existed, it will get overwritten.
+        for (pub_key, validator_cfg) in &self.validators {
+            if validator_cfg.stake == U512::zero() {
+                let balance = validator_cfg.maybe_balance.unwrap_or_else(U512::zero);
+                let _ = self.state_tracker.create_account(pub_key.clone(), balance);
+            }
+        }
+    }
+
+    /// Gets the account for the given public key from the global state, or creates one if it
+    /// doesn't exist and fills it with the amount of tokens specified in the validators config.
+    fn get_or_create_account(&mut self, pub_key: &PublicKey) -> &Account {
+        match self.accounts.entry(pub_key.clone()) {
+            Entry::Vacant(vac) => {
+                let account_hash = pub_key.to_account_hash();
+                let account = match self.builder.get_account(account_hash) {
+                    Some(account) => account,
+                    None => self.state_tracker.create_account(
+                        pub_key.clone(),
+                        self.validators
+                            .get(pub_key)
+                            .and_then(|vcfg| vcfg.maybe_balance)
+                            .unwrap_or_else(U512::zero),
+                    ),
+                };
+                vac.insert(account)
+            }
+            Entry::Occupied(occupied) => occupied.into_mut(),
```

Comment:
Because that's the only method that can convert an `OccupiedEntry` into a reference. The mutable reference will then implicitly get converted to an immutable one.

Reply:
I see, thanks :)